lib/Pipes/RevngPasses.cpp

```diff
@@ -0,0 +1,49 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Model/LoadModelPass.h"
+#include "revng/Pipeline/Context.h"
+#include "revng/Pipeline/ExecutionContext.h"
+#include "revng/Pipeline/LLVMKind.h"
+#include "revng/Pipes/RevngPasses.h"
+#include "revng/Pipes/TaggedFunctionKind.h"
+#include "revng/Support/FunctionTags.h"
+
+bool pipeline::RevngModulePass::runOnModule(llvm::Module &Module) {
+  auto &Analysis = getAnalysis<pipeline::LoadExecutionContextPass>();
+  ExecutionContext *Ctx = Analysis.get();
+  bool Result = false;
+  if (Ctx) {
+    auto name = Analysis.getWrittenContainerName();
+    auto *RequestedTargets = &Ctx->getCurrentRequestedTargets(name);
+    Result = runOnTargets(Module, RequestedTargets);
+    Ctx->getContext().pushReadFields();
+    for (auto &Target : *RequestedTargets)
+      Ctx->commit(Target, Analysis.getWrittenContainerName());
+
+    Ctx->getContext().popReadFields();
+    Ctx->getContext().clearAndResume();
+
+  } else {
+    Result = runOnTargets(Module, nullptr);
```

Comment:
What's the use case for *not* having a context?

Reply:
forgot to remove the if from the version with opt support. Dropped now.